src/Utils.cpp

```diff
@@ -499,6 +499,82 @@ CString CUtils::FormatTime(time_t t, const CString& sFormat,
     return s;
 }
 
+/* Supports additional format specifiers for formatting sub-second values:
+ *
+ * - %L - millisecond, 3 digits (ruby extension)
+ * - %N - sub-second fraction (ruby extension)
+ *   - %3N - millisecond
+ *   - %6N - microsecond
+ *   - %9N - nanosecond (default if no digit specified)
+ *
+ * However, note that timeval only supports microsecond precision
+ * (thus, formatting with higher-than-microsecond precision will
+ * always result in trailing zeroes), and IRC server-time is specified
+ * in millisecond precision (thus formatting received timestamps with
+ * higher-than-millisecond precision will always result in trailing
+ * zeroes).
```

Comment:
So, is there any reason to make nanosecond precision the default? Or even a supported option?

Reply:
The reason is to stick to an existing formatting syntax instead of proliferating them further by diverging or inventing our own.

Reply:
Why do you want at least millisecond precision for format string?